### PR TITLE
introduce message, fix app props key type

### PR DESCRIPTION
**Pull request by nayato**

### CODE REVIEW

**Comment by nayato** [comment]
first (modest) step in support for message level abstractions.

File: build.rs [hunk 1]
```diff
@@ -95,7 +95,11 @@ fn reformat_file(path: &std::path::Path) {
         .expect("failed to open file");
     let mut data = String::new();
     f.read_to_string(&mut data).expect("failed to read file.");
-    data = data.replace("\n\n", "\n")
+    data = data
+        .replace("\r\n\r\n", "\r\n")
+        .replace("\r\n\r\n", "\r\n")
+        .replace("\r\n\r\n", "\r\n")
+        .replace("\n\n", "\n")
```
Comment (myagley) [inline build.rs:102]: What is this for?

File: src/transport/message.rs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        }
+        if let Some(da) = self.delivery_annotations {
+            Section::DeliveryAnnotations(da).encode(&mut dst);
+        }
+        if let Some(ma) = self.message_annotations {
+            Section::MessageAnnotations(ma).encode(&mut dst);
+        }
+        if let Some(p) = self.properties {
+            Section::Properties(p).encode(&mut dst);
+       }
```
Comment (myagley) [inline src/transport/message.rs:42]: nit: whitespace

**Review by myagley — APPROVED** [review]
Just have the two nits.

File: build.rs (hunk 1 above)
Comment (nayato) [reply build.rs:102]: changes in rustfmt (nightly vs stable) change line ending char sequence it seems. Had to add it to account for both. technically we can have an ugly definitions.rs checked in but I'd rather have it formatted to be able to check the code there.

File: src/transport/message.rs (hunk 2 above)
Comment (nayato) [reply src/transport/message.rs:42]: mmm?

Comment (myagley) [reply src/transport/message.rs:42]: The brace doesn't line up with the `if` two lines above.

**Pull request merged**